include/seastar/core/estimated_histogram.hh

```diff
@@ -0,0 +1,385 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2023 ScyllaDB.
+ */
+
+#pragma once
+
+#include <cmath>
+#include <algorithm>
+#include <vector>
+#include <chrono>
+#include <seastar/core/metrics_types.hh>
+#include <seastar/core/print.hh>
+#include <seastar/core/bitops.hh>
+#include <limits>
+#include <array>
+
+namespace seastar {
+namespace metrics {
```

Comment:
it's public for users of the metrics layer

Reply:
we can use `namespace seastar::metrics { ... }` since C++17, IMHO, it's more readable than the nested notation. see syntax (8) in https://en.cppreference.com/w/cpp/language/namespace.

Reply:
I know it's public, but should it be? Making it public means we can't change it later.